### PR TITLE
task: improve error logs

### PR DESCRIPTION
**Pull request by luca-gr4vy**
# Description

Send any `beforeTransactionError` error data to embed-ui, so it can be exposed in the `onEvent - transactionFailed` log info.

**Ticket:** https://gr4vy.atlassian.net/browse/TA-7168

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn lint` to make sure my changes pass all tests
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch
- [ ] I have tested both the react and the CDN versions on local and integration environments
- [x] I have added the necessary labels to this PR in case a new release needs to be published after merging into `main` (e.g. `release` and `patch`)

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: packages/embed/src/index.ts [hunk 1]
```diff
@@ -266,9 +266,10 @@ export function setup(setupConfig: SetupConfig) {
             ]),
           })
         })
-        .catch(() => {
+        .catch((error) => {
           dispatch({
             type: 'beforeTransactionError',
+            data: { beforeTransactionError: error },
```
Comment (douglaseggleton) [inline packages/embed/src/index.ts:272]: This should be just the payload as the event name is already defined.

Screenshot 2024-06-01 at 10 56 40 AM


```suggestion
            data:  error,
```

Comment (douglaseggleton) [reply packages/embed/src/index.ts:272]: After looking at the Embed UI PR I can see why you have done it this way - I think this is okay considering the outcome we need :-)

**Review by douglaseggleton — APPROVED** [review]
LGTM 🚚

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v2.27.1`](https://github.com/gr4vy/gr4vy-embed/releases/tag/v2.27.1) :rocket:

**Pull request merged**